Library/Homebrew/tap.rb

```diff
@@ -280,4 +316,93 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
```

Comment:
> This makes me a bit nervous. I'm wondering if it's worth having the tap check out into those locations. There will likely be older tools around Homebrew than make assumptions about these paths (and we've said "the filesystem is our API" for a long time).

We can replace them to symlinks.

>  I also wonder if it's worth considering moving extracting core into a new repository too; this repository is pretty massive at this point particularly if you pull all e.g. PR heads. This could be a later stage

Sadly we can't due to current `update` method. One goal I want to ensure is that `brew update` should pass on all of existed Travis CI OS X image.

> except I do think it should be able to be uninstalled (or at least another tap chosen somehow as the "primary" tap). This would be useful for Linuxbrew and other users of the Homebrew core code (which we can actually apply semver to and perhaps even turn into a RubyGem).

This certainly can be achieved, but it's more sense to be handled internally by maintainers rather than average users.

> I'm :ok: with this particularly if it's declared in the same file.
> I think it's definitely fine to rename internal classes that we mark as @private.

I would still prefer the inheritance approach, so I will rename the class then.